src/Spago/Config/AST.hs

```diff
@@ -0,0 +1,964 @@
+                                updateListTextByWrappingListAppend additions
+                                  $ Dhall.Field var (Dhall.makeFieldSelection key)
+
+                              -- `{ irrelevantKey = "irrelevantValue", key = __embed.dependencies # ["new"] }`
+                              newRight =
+                                Dhall.RecordLit
+                                $ flip (Dhall.Map.insert key) kvs
+                                $ Dhall.makeRecordField lsAppend
+                            pure $ Just $ Updated $ Dhall.Let binding (Dhall.Prefer charSet preferAnn var newRight)
+
+                          --    `./spago.dhall // { x = { irrelevantKey = "irrelevantValue" } }.x`
+                          -- to
+                          --    `( let __embed = ./spago.dhall
+                          --       in  __embed with dependencies = __embed.dependencies # ["name"]
+                          --     ) // { x = { irrelevantKey = "irrelevantValue" } }.x`
+                          _ -> do
+                            pure $ Just $ Updated
+                              $ Dhall.Prefer charSet preferAnn (updateListTextByWrappingLetBinding (key :| []) additions "__embed" left) right
+
+                      SetText t -> do
+                        case right of
+                          --    `./spago.dhall // { irrelevantKey = "irrelevantValue" }`
+                          -- to
+                          --    `./spago.dhall // { irrelevantKey = "irrelevantValue", name = "new" }`
+                          Dhall.RecordLit kvs -> do
+                            let
+                              newRight =
+                                Dhall.RecordLit
+                                $ flip (Dhall.Map.insert key) kvs
+                                $ Dhall.makeRecordField t
+                            pure $ Just $ Updated $ Dhall.Prefer charSet preferAnn left newRight
+
+                          --    `./spago.dhall // { x = { irrelevantKey = "irrelevantValue" } }.x`
+                          -- to
+                          --    `./spago.dhall with name = "name" // { x = { irrelevantKey = "irrelevantValue" } }.x`
+                          _ -> do
+                            pure $ Just $ Updated $ Dhall.Prefer charSet preferAnn (Dhall.With left (key :| []) t) right
+
+                  varName@(Just (VariableName _)) -> do
+                    -- `The `Just VariableName` can't happen here because this is `AtRootExpression`
+                    pure varName
+
+                  Just (Updated newLeft) -> do
+                    pure $ Just $ Updated $ Dhall.Prefer charSet preferAnn newLeft right
+
+                  Nothing -> pure Nothing
+
+          SearchingForField _ -> do
+            -- See Dhall.Prefer's `AtRootExpression` case
+            -- but for this situation, we're trying to find a record, so we don't match against a "dependencies" field
+            mpRight <- updateExpr level right
```

Comment:
What's `mp`?

Reply:
Typo! That's supposed to be `mb`. And yes, I'll change this to `maybeRight`.

Reply:
Done.